Dockerfile

```diff
@@ -34,24 +37,33 @@ chown -R root:root /opt/pseudos/;                                  \
 chmod -R +r /opt/pseudos/
 
 # Install Python packages needed for AiiDA lab.
-RUN pip install                    \
-    'aiidalab==v20.05.0b1'         \
+RUN pip install 'aiidalab==v20.05.0b1'
+
+# Installing Jupyter-related things in the root environment.
+RUN /usr/bin/pip3 install          \
     'jupyterhub==0.9.4'            \
     'jupyterlab==0.35.4'           \
-    'nbserverproxy==0.8.8'
-
-# Enable nbserverproxy extension.
-RUN jupyter serverextension enable --sys-prefix --py nbserverproxy
+    'fileupload==0.1.5'            \
+    'nbserverproxy==0.8.8'         \
+    'appmode-aiidalab==0.5.0.1'    \
+    'notebook==5.7.8'              \
+    'nglview'                      \
+    'voila'
+
+# Enable extensions.
+RUN /usr/local/bin/jupyter serverextension enable --sys-prefix --py nbserverproxy
+RUN /usr/local/bin/jupyter nbextension     enable --py --sys-prefix appmode
```

Comment:
also here: why is this needed now, but wasn't needed before?

Reply:
Well, without this appmode won't work. Not sure why, though.

The docker container that Ole had created actually contained that line. His container was based on the same principles we are now basing the new container on - keeping Jupyter stuff in a separate python environment.

When I was moving stuff to AiiDA 1.0, I removed it because I moved everything in the same Python3 environment. Now I am moving things back just to give another evidence that _history repeats itself_ ;)